src/java/io/bazel/rulesscala/scalac/CompileOptions.java

```diff
@@ -36,12 +37,17 @@ public CompileOptions(List<String> args) {
 
     scalaOpts = getCommaList(argMap, "ScalacOpts");
     printCompileTime = booleanGetOrFalse(argMap, "PrintCompileTime");
+    expectJavaOutput = booleanGetOrTrue(argMap, "ExpectJavaOutput");
     pluginArgs = buildPluginArgs(getOrEmpty(argMap, "Plugins"));
     classpath = getOrError(argMap, "Classpath", "Must supply the classpath arg");
     files = getCommaList(argMap, "Files");
 
     javaFiles = getCommaList(argMap, "JavaFiles");
 
+    if (!expectJavaOutput && javaFiles.length != 0) {
+      throw new RuntimeException("Cannot hava java source files when no expected java output");
```

Comment:
@ianof sorry for the late eyes but aren’t you missing a test for this case? IIUC this is where java files are on srcs “regularly” (not in srcjar)

Reply:
there is a test for this case: https://github.com/bazelbuild/rules_scala/pull/529/commits/1d006810639378c52309adb5607c835dc2346d7d#diff-9e850e1e0893c9ea027f87c0b45dffabR188